PerfTools/AllocMonitor/plugins/ThreadTracker.cc

```diff
@@ -0,0 +1,31 @@
+// -*- C++ -*-
+//
+// Package:     PerfTools/AllocMonitor
+// Class  :     ThreadTracker
+//
+// Implementation:
+//     [Notes on implementation]
+//
+// Original Author:  Christopher Jones
+//         Created:  Mon, 11 Nov 2024 22:54:23 GMT
+//
+
+// system include files
+
+// user include files
+#include "ThreadTracker.h"
+
+//
+// constants, enums and typedefs
+//
+
+//
+// static data member definitions
+//
+
+namespace allocMon {
+  ThreadTracker& ThreadTracker::instance() {
+    static ThreadTracker s_tracker;
```

Comment:
Maybe
```suggestion
    CMS_THREAD_SAFE static ThreadTracker s_tracker;
```
would be worth it (although I see static analyzer warning many `static`s, so perhaps a dedicated PR to annotate all of them would make sense)

Reply:
done